src/EnergyPlus/DataSystemVariables.cc

```diff
@@ -251,81 +240,51 @@ namespace DataSystemVariables {
-            return;
-        } else {
-            print(state.files.audit, "{}={}\n", "not found (input file)", getAbsolutePath(DataStringGlobals::inputDirPathName + InputFileName));
-        }
-
-        // Look relative to input path
-        if (FileSystem::fileExists(envinputpath1 + InputFileName)) {
-            FileFound = true;
-            CheckedFileName = envinputpath1 + InputFileName;
-            print(state.files.audit, "{}={}\n", "found (epin)", getAbsolutePath(CheckedFileName));
-            return;
-        } else {
-            print(state.files.audit, "{}={}\n", "not found (epin)", getAbsolutePath(envinputpath1 + InputFileName));
-        }
-
-        // Look relative to input path
-        if (FileSystem::fileExists(envinputpath2 + InputFileName)) {
-            FileFound = true;
-            CheckedFileName = envinputpath2 + InputFileName;
-            print(state.files.audit, "{}={}\n", "found (input_path)", getAbsolutePath(CheckedFileName));
-            return;
-        } else {
-            print(state.files.audit, "{}={}\n", "not found (input_path)", getAbsolutePath(envinputpath2 + InputFileName));
-        }
-
-        // Look relative to program path
-        if (FileSystem::fileExists(envprogrampath + InputFileName)) {
-            FileFound = true;
-            CheckedFileName = envprogrampath + InputFileName;
-            print(state.files.audit, "{}={}\n", "found (program_path)", getAbsolutePath(CheckedFileName));
-            return;
-        } else {
-            print(state.files.audit, "{}={}\n", "not found (program_path)", getAbsolutePath(envprogrampath + InputFileName));
-        }
-
-        if (!TestAllPaths) return;
-
-        // Look relative to current working folder
-        if (FileSystem::fileExists(CurrentWorkingFolder + InputFileName)) {
-            FileFound = true;
-            CheckedFileName = CurrentWorkingFolder + InputFileName;
-            print(state.files.audit, "{}={}\n", "found (CWF)", getAbsolutePath(CheckedFileName));
-            return;
-        } else {
-            print(state.files.audit, "{}={}\n", "not found (CWF)", getAbsolutePath(CurrentWorkingFolder + InputFileName));
+        std::vector<std::pair<std::string, std::string>> pathsChecked;
+
+        const std::vector<std::pair<std::string, std::string>> pathsToCheck = {
+            {InputFileName, "Current Working Directory"},
+            {DataStringGlobals::inputDirPathName + InputFileName, "IDF Directory"},
+            {DataStringGlobals::exeDirectory + InputFileName, "EnergyPlus Executable Directory"},
+            {envinputpath1 + InputFileName, "\"epin\" Environment Variable"},
+            {envinputpath2 + InputFileName, "\"input_path\" Environment Variable"},
+            {envprogrampath + InputFileName, "\"program_path\" Environment Variable"},
+            {CurrentWorkingFolder + InputFileName, "INI File Directory"},
+            {ProgramPath + InputFileName, "\"program\", \"dir\" from INI File"}
+        };
```

Comment:
This is quite a small list, so it's not a big deal, but an array would've been acceptable here since I think this vector is staying fixed length.